CodeEdit/ContentView.swift

```diff
@@ -20,64 +20,107 @@ struct MainContentView: View {
+    
+    func openFolderDialog() {
+        let dialog = NSOpenPanel()
+        
+        dialog.title = "Select a Folder to Open"
+        dialog.allowsMultipleSelection = false
+        dialog.canChooseFiles = false
+        dialog.canChooseDirectories = true
+        
+        if dialog.runModal() == NSApplication.ModalResponse.OK {
+            if let result = dialog.url {
+                do {
+                    workspace = try Workspace(url: result)
+                } catch {
+                    alertTitle = "Unable to Open Folder"
+                    alertMsg = error.localizedDescription
+                    showingAlert = true
+                    print(error.localizedDescription)
+                }
+            }
+        }
+    }
+    
+    func openFileEditor(path: URL) {
+        // TODO: I can't seem to figure this one out
+    }
 
     var body: some View {
         NavigationView {
-            VStack {
-                List(selection: $selection) {
-                    NavigationLink(
-                        destination: MainContentView()
-                            .navigationTitle("Home")
-                            .navigationSubtitle("Dashboard"),
-                        label: {
-                            Image(systemName: "house")
-                                .foregroundColor(.secondary)
-                            Text("Home")
-                        }
-                    ).id("home")
-                    Section(header: Text("Items")) {
-                        ForEach(items, id: \.self) { item in
-                            NavigationLink(
-                                destination: Text("Item \(item)")
-                                    .navigationTitle("Item \(item)"),
-                                label: {
-                                    Image(systemName: "folder")
-                                        .foregroundColor(.secondary)
-                                    Text("Item \(item)")
+            List {
+                if let workspace = workspace {
+                    Section(header: Text(workspace.directoryURL.lastPathComponent)) {
+                        ForEach(workspace.directoryContents, id: \.absoluteURL) { url in
+                            let fileManager = FileManager.default
+                            let filePath = url.path
+                            var isDir: ObjCBool = false
+                            
+                            if fileManager.fileExists(atPath: filePath, isDirectory: &isDir) {
+                                if isDir.boolValue {
+                                    Label(url.lastPathComponent, systemImage: "folder.fill")
+                                } else {
+                                    Button(action: { openFileEditor(path: url) }) {
+                                        Label(url.lastPathComponent, systemImage: "doc.fill")
+                                    }
+                                    .buttonStyle(.plain)
```

Comment:
Just a small note to be considered. I believe we can use `OutlineGroup` here with something like `FileItem` structure as below so that we will be able to have folder hierarchy UI.

```swift
struct FileItem: Hashable, Identifiable {
    var id: UUID = UUID()
    var url: URL
    var children: [FileItem]? = nil
    var systemImage: String {
        switch children {
        case nil:
            return "doc.plaintext"
        case .some(let children):
            return children.isEmpty ? "folder" : "folder.fill"
        }
    }
}

struct ContentView {
    ...

    var sidebar: some View {
        List {
            OutlineGroup(project.fileItems, children: \.children) { item in
                Button(action: {openFileEditor(path: url)}) {
                    Label(item.url.lastPathComponent, systemImage: item.systemImage)
                        .accentColor(.gray)
                        .font(.callout)
                }.buttonStyle(.plain)
            }
            
        }
        .frame(minWidth: 250)
        .listStyle(.sidebar)
    }
}
```

But also we need to consider fetching subdirectories' content only when needed because it would be a resource-intensive task to index all files if we are speaking about cache directories, etc.

Reply:
I've added these changes + fetching subfolders and files in the latest commit. Check it out and lemme know what you think.

Reply:
Looks awesome! 🤩